public/app/plugins/datasource/grafana-azure-monitor-datasource/components/DefaultSubscription.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { FunctionComponent, useEffect, useReducer } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { InlineField, Select, Button } from '@grafana/ui';
+
+import { isCredentialsComplete } from '../credentials';
+import { selectors } from '../e2e/selectors';
+import { AzureCredentials, AzureDataSourceJsonData } from '../types';
+
+const LABEL_WIDTH = 18;
+
+interface Props {
+  options: AzureDataSourceJsonData;
+  credentials: AzureCredentials;
```

Comment:
`DefaultSubscripton` doesn't actually need to know anything about credentials.

In my opinion, it could be implemented with only three main properties:
- `subscriptions` - the list of subscriptions to select from (supplied from outside and empty if not yet loaded)
- `subscriptionId`\`selectedSubscription` - the subscription which was actually selected, `undefined` if no subscription has been selected.
- `onSubscriptionChange` - callback when another subscription being selected, so the outside code, `MonitorConfig` can apply it to datasource settings.

Every time `AzureCredentialsForm` changes credentials, it calls` onCredentialsChange` and `MonitorConfig` clears the `subscriptions` state variable which would propagate to the `DefaultSubscription` control.

`MonitorConfig` could also be responsible for the asynchronous call to fetch the list of subscriptions since it knows when credentials change and knows when to re-fetch the subscriptions.